src/macros.rs

```diff
@@ -0,0 +1,13 @@
+/// Shorthand for unwrap_or_else to provide context for panics.
+///
+/// The macro should be used similar to `assert!` with an intermediary `maybe_value` or `result_value`.
+///
+/// ```ignore
+/// let maybe_value = my_values.iter().find(|x| == 42);
+/// let value = assume!(maybe_value, "The value should be defined because: {:?}", something);
+/// ```
```

Comment:
Worth expanding on why we want to use `unwrap_or_else` and the limitation on only being usable with Option/Result

Reply:
True that